marconi-mamba/src/Marconi/Api/UtxoIndexersQuery.hs

```diff
@@ -7,94 +7,166 @@ module Marconi.Api.UtxoIndexersQuery
-import Data.Text (Text, unpack)
+import Data.Set (Set, fromList, toList)
+import Data.Text (Text, intercalate, unpack)
 import Database.SQLite.Simple (NamedParam ((:=)), open)
 import Database.SQLite.Simple qualified as SQL
 import Ledger (Address, TxOutRef)
 import Ledger.Tx.CardanoAPI (ToCardanoError (DeserialisationError, Tag), fromCardanoAddress)
 import Marconi.Api.Types (CardanoAddress, DBConfig (DBConfig),
-                          DBQueryEnv (DBQueryEnv, _dbConf, _queryAddresses, _queryQSem), HasDBConfig (utxoConn),
-                          HasDBQueryEnv (dbConf, queryQSem, queryQSem), TargetAddresses,
-                          UtxoRowWrapper (UtxoRowWrapper))
+                          DBQueryEnv (DBQueryEnv, _dbConf, _network, _queryAddresses, _queryQSem),
+                          HasDBConfig (utxoConn), HasDBQueryEnv (dbConf, queryAddresses, queryQSem, queryQSem),
+                          TargetAddresses, UtxoRowWrapper (UtxoRowWrapper), UtxoTxOutReport (..))
 import Marconi.Index.Utxo (UtxoRow (UtxoRow, _reference))
 
 bootstrap
     ::  FilePath
     -> TargetAddresses
+    -> CApi.NetworkId
     -> IO DBQueryEnv
-bootstrap dbPath targetAddresses = do
+bootstrap dbPath targetAddresses nId = do
     dbconf <- DBConfig <$> open dbPath
     qsem <- newQSemN 1
     pure $ DBQueryEnv
         {_dbConf = dbconf
         , _queryQSem = qsem
-        , _queryAddresses = targetAddresses}
+        , _queryAddresses = targetAddresses
+        , _network = nId
+        }
+
+findAll
+    :: DBQueryEnv
+    -> IO (Set UtxoTxOutReport)    -- ^ set of corresponding TxOutRefs
+findAll env =
+    withQueryAction (findAll' env) (env ^. queryQSem)
+
+findAll'
+    :: DBQueryEnv
+    -> IO (Set UtxoTxOutReport)    -- ^ set of corresponding TxOutRefs
+findAll' env = fromList <$> forConcurrently addresses f
+    where
+        addresses = NonEmpty.toList (env ^. queryAddresses)
+        f  :: CardanoAddress -> IO (UtxoTxOutReport)
+        f addr = UtxoTxOutReport (CApi.serialiseAddress addr) <$> findByCardanoAddress' env addr
 
 findByPlutusAddress
     :: DBQueryEnv
     -> Address              -- ^ Plutus address
     -> IO (Set TxOutRef)    -- ^ set of corresponding TxOutRefs
-findByPlutusAddress env address = withQueryAction (env ^. queryQSem) action
-    where
-        action = (fromList <$> SQL.queryNamed
-                    (env ^. dbConf . utxoConn)
-                    "SELECT txId FROM utxos WHERE utxos.address=:address"
-                    [":address" := address])
+findByPlutusAddress env address =
+    withQueryAction (findByPlutusAddress' env address) (env ^. queryQSem)
+
+findByPlutusAddress'
+    :: DBQueryEnv
+    -> Address              -- ^ Plutus address
+    -> IO (Set TxOutRef)
+findByPlutusAddress' env address =
+    SQL.queryNamed (env ^. dbConf . utxoConn)
+                  "SELECT * FROM utxos WHERE utxos.address=:address"
+                  [":address" := address]
+    >>= pure . fromList . fmap _reference
```

Comment:
That's not using Marconi query interface is it? This only accesses data that is on disk. However, there's data that can be in memory.

Reply:
Yes, it is missing the in-memory data that is yet to be written to the disk, which I totally missed when I switched form hot-store to cold-store(SQLite) I'll add those in.